src/GitStatusCache/src/Git.cpp

```diff
@@ -267,9 +268,46 @@ bool Git::GetRefStatus(Git::Status& status, UniqueGitRepository& repository)
 	result = git_branch_upstream(&upstream.get(), head.get());
 	if (result == GIT_ENOTFOUND)
 	{
-		Log("Git.GetRefStatus.NoUpstream", Severity::Spam)
-			<< R"(Branch does not have a remote tracking reference. { "repositoryPath": ")" << status.RepositoryPath
-			<< R"(", "localBranch": ")" << status.Branch << R"(" })";
+		auto upstreamRemoteName = git_buf{ 0 };
+		auto upstreamRemoteResult = git_branch_upstream_remote(

```

Comment:
I don't think you need this now - `git_branch_upstream_name` includes a remote check.